Mathlib/LinearAlgebra/CliffordAlgebra/SpinGroup.lean

```diff
@@ -0,0 +1,474 @@
+/-
+Copyright (c) 2022 Jiale Miao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jiale Miao, Utensil Song, Eric Wieser
+-/
+import Mathlib.GroupTheory.GroupAction.ConjAct
+import Mathlib.Algebra.Star.Unitary
+import Mathlib.LinearAlgebra.CliffordAlgebra.Star
+import Mathlib.LinearAlgebra.CliffordAlgebra.Even
+import Mathlib.LinearAlgebra.CliffordAlgebra.Inversion
+
+/-!
+# The Pin group and the Spin group
+
+In this file we define `lipschitzGroup`, `pinGroup` and `spinGroup` and show they form a group.
+
+## Main definitions
+
+* `lipschitzGroup`: the Lipschitz group with a quadratic form.
+* `pinGroup`: the Pin group defined as the infimum of `lipschitzGroup` and `unitary`.
+* `spinGroup`: the Spin group defined as the infimum of `pinGroup` and `CliffordAlgebra.even`.
+
+## Implementation Notes
+
+Here are some discussion about the latent ambiguity of definition :
+https://mathoverflow.net/q/427881/172242 and https://mathoverflow.net/q/251288/172242
+
+The definition of the Lipschitz group `{𝑥 ∈ 𝐶𝑙(𝑉,𝑞) │ 𝑥 𝑖𝑠 𝑖𝑛𝑣𝑒𝑟𝑡𝑖𝑏𝑙𝑒 𝑎𝑛𝑑 𝑥𝑣𝑥⁻¹∈ 𝑉}` is given by:
```

Comment:
This use of unicode is cute, but we should just use LaTeX for the benefit of readers of the web documentation.

Reply:
Done, but there are 2 issues:

1. the readability is reduced for source code readers
2. doc-gen4 still has issues for LaTeX in markdown, in this case, a valid `\{` has to be `\\{` for the `{` of the set to rendered correctly, similar LaTeX formulas are long broken in Mathlib 4 doc, [previous discussions](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/LaTeX.20problem.20in.20docstring) suggested that we should not fix the LaTeX formulas and should wait for the fix from doc-gen4 parser but it's been too long now, and I can't PR a broken LaTeX, so I have to fix it here, but it will be broken again once it's fixed on the parser end, either by doc-gen4 or verso.